fsw/src/crypto_props.c

```diff
@@ -0,0 +1,351 @@
+                        //printf("DEBUG current_line DATA=%s\n",current_line);
+                        //1) pass the key 
+                        int i=0; 
+                        while (i<index)
+                        {
+                            key[i]=current_line[i];
+                            i++; 
+                        }
+                        key[i] = '\0';
+                        //2) parse the value 
+                        i=index+1;
+                        int j=0;
+                        while (current_line[i]!='\0' && j<p_self->VALUE_SIZE_IN_BYTES)
+                        {
+                            val[j]=current_line[i];
+                            i++;
+                            j++;
+                        }
+                        val[j] = '\0'; 
+                        //printf("DEBUG key=%s,value=%s\n",key,val);
+                        //3) add the key,value to the properties linkedlist
+                        crypto_config_props_add_property_helper(key,val,&p_self);
+                        
+                    }
+                    else
+                    {   //delimiter was detected but this starts with a comment character therefore it will NOT be parsed
+                        //printf("COMMENT=%s\n",current_line);
+                    }
+                }
+                else
+                {
+                    //do nothing there was no "=" delimiter detected 
+                }
+                
+            }
+            if (NULL!=p_file)
+            {
+                 fclose(p_file);
+            }
+        }
+        else
+        {
+            //error 
+            printf("ERROR, crypto_config_load_properties() error opening file!.\n");
+        }
+
+    }
+    else
+    {
+        printf("ERROR, crypto_config_load_properties() file path is NULL.\n");
+    }
+}
+
+void crypto_config_print_all_props(crypto_config_list* p_self)
+{
+    if (NULL!=p_self)
+    {
+         crypto_config_node* p_current = p_self->p_head;
+         while (NULL!=p_current)
+         {
+             printf("key=%s,value=%s\n",p_current->key,p_current->value);
+             p_current = p_current->p_next; 
+             
+         }
+    }
+}
+
+/*===========================================================================
+Function:           safe_copy
+Description:        safe_copy is a helper function to guard against unsafe copies.
+ *                  strlcp() is not available in all compilers, therefore this 
+                    function returns true(1) if destination size >= src.   
+Inputs:             dest - string 
+ *                  src - string                   
+Outputs:            1 or 0 
+References: 
+==========================================================*/
+int safe_copy(char* dest, char* src)
+{
+    int flag = 0; 
+    if (NULL!=dest && NULL!=src)
+    {
+        if (sizeof(dest)>=sizeof(src))
+        {
+            flag = 1; 
+        }
+    }
+    return flag; 
+}
+
```

Comment:
Hmmm... I might be missing something, but safe_copy seems to be a pointless function? sizeof(dest) is always just 8 bytes since it's a char pointer, this shouldn't actually be comparing the length of each string (which is seems like is the intention of this function?).

Seems like this function and all calls to it should be entirely deleted.

Reply:
Yes, good catch. This function was updated. I need a way to test against buffers overwrites & empty chars return 0 length even with strlen(). The new function was tested against many values.